### PR TITLE
Clarify Annotation arrowprops docs

### LINKED ISSUE #21733: [Bug]: Possible bug on arrows in annotation
### Bug summary

Arrow behavior in `axes.annotate` is inconsistent depending on the direction of the xy coordinate.

Arrow line begins at center of text in cases of horizontal/vertical lines, but begins at lower left of text in cases of diagonal lines.

### Code for reproduction

```python
import matplotlib.pyplot as plt

fig = plt.figure()
ax = fig.add_subplot(1,1,1)
ax.set_xlim(0,1)
ax.set_ylim(0,1)
ax.set_aspect('equal')
ax.grid(True)
for target in ((0.2,0.2),(0.2,0.4),(0.2,0.6),(0.4,0.2),(0.6,0.2)):
    ax.annotate('Blah! Blah blah blah.',xy=target, xytext=(0.6,0.6),
                 bbox=dict(boxstyle='round', facecolor='none'),
                 ha='center',va='center',
                 arrowprops=dict(facecolor='black',width=0.8,headwidth=5,connectionstyle="arc3,rad=0"))
```

### Actual outcome:

matplotlib2

### Expected outcome

Arrow starting point should be consistent, or at least adjustable; relpos is not available as a FancyArrowPatch key, so if I add it, I get `AttributeError: 'FancyArrowPatch' object has no property 'relpos'`

### Additional information

_No response_

### Operating system

Windows

### Matplotlib Version

import matplotlib; print(matplotlib.__version__)

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

3.7.4

### Jupyter version

6.1.4

### Installation

conda

### PR DESCRIPTION
**Pull request by timhoffm**
Closes #21733.

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -1727,35 +1731,22 @@ def transform(renderer) -> Transform
             ?            Any key to :class:`matplotlib.patches.FancyArrowPatch`
             ==========   ======================================================
 
-            If *arrowprops* contains the key 'arrowstyle' the
-            above keys are forbidden.  The allowed values of
-            ``'arrowstyle'`` are:
-
-            ============   =============================================
-            Name           Attrs
```
Comment (QuLogic) [inline lib/matplotlib/text.py:1735]: Did you drop this table on purpose?

File: lib/matplotlib/text.py [hunk 2]
```diff
@@ -1710,9 +1710,13 @@ def transform(renderer) -> Transform
 
         arrowprops : dict, optional
             The properties used to draw a `.FancyArrowPatch` arrow between the
-            positions *xy* and *xytext*. Note that the edge of the arrow
-            pointing to *xytext* will be centered on the text itself and may
-            not point directly to the coordinates given in *xytext*.
+            positions *xy* and *xytext*.  Defaults to None, i.e. no arrow is
+            drawn.
+
+            For historical reasons there are two different ways of usage, that
```
Comment (QuLogic) [inline lib/matplotlib/text.py:1716]:
```suggestion
            For historical reasons, there are two different usages that
```

Comment (timhoffm) [reply lib/matplotlib/text.py:1716]: I'm not sure that's what I want to say. IMHO there is only one usage, namely specifying the type of arrows, but there are two ways to do it. I was considering using "two different APIs" but thought that is too technical.

File: lib/matplotlib/text.py [hunk 3]
```diff
@@ -1710,9 +1710,13 @@ def transform(renderer) -> Transform
...
+            are distinguised by whether 'arrowstyle' is specified in the dict.
```
Comment (jason-sachs) [inline lib/matplotlib/text.py:1717]: `distinguished`

File: lib/matplotlib/text.py (hunk 2 above)
Comment (jklymak) [reply lib/matplotlib/text.py:1716]: Maybe: 

```suggestion
            For historical reasons there are two different ways to specify arrows, "simple" and "fancy":
```

and maybe drop the next clause?

Comment (timhoffm) [reply lib/matplotlib/text.py]: Followed @jklymak suggestion.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**